### PR TITLE
SRE-1994 s3-http-domain

### PR DESCRIPTION
**Pull request by sathg**
**Summary**
Updated the Amazon S3 Path, as the current s3 HTTP domain is getting [deprecated](https://aws.amazon.com/blogs/aws/amazon-s3-path-deprecation-plan-the-rest-of-the-story/)

**Related Ticket**
https://adroll.atlassian.net/browse/SRE-1994

### CODE REVIEW

File: src/erlcloud_s3.erl [hunk 1]
```diff
@@ -1887,15 +1887,15 @@ aws_region_from_host(Host) ->
...
 %%  redirected to "bucket-frankfurt.s3.amazonaws.com" endpoint.
 %%
-%%  The 2nd ("https://s3.amazonaws.com/?acl" with
+%%  The 2nd ("https://?acl.s3.amazonaws.com" with
 %%  {"host","bucket-frankfurt.s3.amazonaws.com"} header) is redirected
 %%  to "bucket-frankfurt.s3.eu-central-1.amazonaws.com".
 %%
 %%  And finally the 3rd request succeeds -
-%%  ("https://s3.eu-central-1.amazonaws.com/?acl" with
+%%  ("https://?acl.s3.eu-central-1.amazonaws.com" with
```
Comment (rb-nr) [inline src/erlcloud_s3.erl]: the 2 references above look wrong. they should not be changed.

File: test/erlcloud_s3_tests.erl [hunk 2]
```diff
@@ -174,21 +174,21 @@ set_bucket_notification_test_() ->
...
-        fun("https://s3.amazonaws.com/?notification", _, _, _, _, _) -> Response end),
+        fun("https://?notification.s3.amazonaws.com", _, _, _, _, _) -> Response end),
     ?_assertEqual(?S3_BUCKET_EVENTS_LIST_NO_PREFIX,
         erlcloud_s3:get_bucket_attribute("BucketName", notification, config())).
 
 get_bucket_notification_no_suffix_test(_) ->
     Response = {ok, {{200, "OK"}, [], ?S3_BUCKET_EVENT_XML_CONFIG_NO_SUFFIX}},
     meck:expect(erlcloud_httpc, request,
-        fun("https://s3.amazonaws.com/?notification", _, _, _, _, _) -> Response end),
+        fun("https://?notification.s3.amazonaws.com", _, _, _, _, _) -> Response end),
```
Comment (rb-nr) [inline test/erlcloud_s3_tests.erl]: these 3 references should not be changed.

File: src/erlcloud_s3.erl (hunk 1 above)
Comment (sathg) [reply src/erlcloud_s3.erl]: Will revert it back.

File: test/erlcloud_s3_tests.erl (hunk 2 above)
Comment (sathg) [reply test/erlcloud_s3_tests.erl]: Will revert it back.

**Pull request merged**